scheduler/generic_sched_test.go

```diff
@@ -2660,11 +2660,15 @@ func TestServiceSched_NodeDown(t *testing.T) {
 	toBeRescheduled := map[string]bool{allocs[8].ID: true}
 
 	// Mark some allocs as running
-	for i := 0; i < 4; i++ {
+	for i := 0; i < 3; i++ {
 		out := allocs[i]
 		out.ClientStatus = structs.AllocClientStatusRunning
 	}
 
+	// Mark one alloc as in-progress evict
+	allocs[4].DesiredStatus = structs.AllocDesiredStatusEvict
+	allocs[4].ClientStatus = structs.AllocClientStatusRunning
+
 	// Mark appropriate allocs for migration
 	toBeMigrated := map[string]bool{}
 	for i := 0; i < 3; i++ {
```

Comment:
The way the test originally was, we had 4 running allocs and marked 3 for migration, leaving 1 running but not marked for migration. If we use the 4th alloc for our new test w/o changing this loop, we're removing that test case. So we should update this to be `for i := 0; i < 2; i++ {` or even:

```go
allocs[0].DesiredTransition.Migrate = helper.BoolToPtr(true)
allocs[1].DesiredTransition.Migrate = helper.BoolToPtr(true)
toBeMigrated[allocs[0].ID] = true
toBeMigrated[allocs[1].ID] = true
```

(For what it's worth, I sort of wish the original test case was more of a table-driven test; there's no obvious reason to test `allocs[0:3]` if they're all identical. But we can come back to that in a separate PR.)

Reply:
Sounds good, I'll look into switching to a table driven, having the ability to tack on more cases as they arise seems worth while